build fails (cygwin)

no makefile and no configure script, so I tried `cmake .`, then make:

[ 50%] Building C object CMakeFiles/cmatrix.dir/cmatrix.c.o
/home/towo/tmp/cmatrix/cmatrix.c: In function ‘resize_screen’:
/home/towo/tmp/cmatrix/cmatrix.c:277:10: error: lvalue required as left operand of assignment
  277 |     COLS = win.ws_col;
      |          ^
/home/towo/tmp/cmatrix/cmatrix.c:278:11: error: lvalue required as left operand of assignment
  278 |     LINES = win.ws_row;
      |           ^
/home/towo/tmp/cmatrix/cmatrix.c:282:15: error: lvalue required as left operand of assignment
  282 |         LINES = 10;
      |               ^
/home/towo/tmp/cmatrix/cmatrix.c:285:14: error: lvalue required as left operand of assignment
  285 |         COLS = 10;
      |              ^
/home/towo/tmp/cmatrix/cmatrix.c: In function ‘main’:
/home/towo/tmp/cmatrix/cmatrix.c:560:41: error: ‘TIOCSTI’ undeclared (first use in this function)
  560 |                     ioctl(STDIN_FILENO, TIOCSTI, (char*)(str + i));
      |                                         ^~~~~~~

--- a/cmatrix.c
+++ b/cmatrix.c
@@ -69,6 +69,10 @@
 #include <termio.h>
 #endif
 
+#ifdef __CYGWIN__
+#define TIOCSTI 0x5412
+#endif
+
 /* Matrix typedef */
 typedef struct cmatrix {
     int val;

